cli/module_loader.rs

```diff
@@ -458,12 +469,50 @@ impl CliModuleLoader {
       return Err(generic_error("Attempted to load JSON module without specifying \"type\": \"json\" attribute in the import statement."));
     }
 
+    let code_cache = if module_type == ModuleType::JavaScript {
+      self.shared.code_cache.as_ref().and_then(|cache| {
+        let code_hash = self
+          .shared
+          .module_info_cache
+          .get_module_source_hash(specifier, code_source.media_type)
+          .ok()
+          .flatten();
+        let code_timestamp = specifier_to_file_path(specifier)
+          .ok()
+          .and_then(|path| std::fs::metadata(path).ok())
+          .and_then(|m| m.modified().ok())
+          .and_then(|t| t.duration_since(UNIX_EPOCH).ok())
+          .map(|d| d.as_millis() as u64);
```

Comment:
Let's extract this to a function that uses `?` for early exit and use it here and below in `code_cache_ready`. Something like:

```
fn code_hash_key(specifier: ...) -> Result<(hash_type, u64)> {
  let hash = ...?.flatten();
  let path = specifier_to_file_path(specifier)?;
  let timestamp = std::fs::metadata(path)?.modifier()?.duration_since(UNIX_EPOCH)?;
  Ok((hash, path))
}
```